cmd/flags_misc.go

```diff
@@ -201,5 +208,18 @@ func ParseMiscFlags(ctx context.Context, cmd *cobra.Command) (context.Context, e
 		ctx = WithNodeOptions(ctx, nodebuilder.WithMetrics(opts, NodeType(ctx)))
 	}
 
+	ok, err = cmd.Flags().GetBool(p2pMetrics)
+	if err != nil {
+		panic(err)
```

Comment:
return err (and would be nice to return err everywhere in the function instead of panicking.

Reply:
The reason we panic here and in other places like that is that it errors our only if such flag not exists. And if it's not, than it was not added, which is a programmer mistake and should panic imo